pkg/spoke/submarineragent/connections_controller.go

```diff
@@ -79,20 +85,70 @@ func (c *connectionsStatusController) sync(ctx context.Context, syncCtx factory.
 	}
 
 	// check submariner agent status and update submariner-addon status on the hub cluster
-	condition := c.checkSubmarinerConnections(submariner)
+	gatewaycondition := c.checkSubmarinerConnections(submariner)
 
-	updatedStatus, updated, err := addon.UpdateStatus(ctx, c.addOnClient, c.clusterName, addon.UpdateConditionFn(condition))
+	updatedStatus, updated, err := addon.UpdateStatus(ctx, c.addOnClient, c.clusterName, addon.UpdateConditionFn(gatewaycondition))
 	if err != nil {
 		return err
 	}
 
 	if updated {
-		c.logger.Infof("Updated submariner ManagedClusterAddOn status condition: %s", resource.ToJSON(condition))
+		c.logger.Infof("Updated submariner ManagedClusterAddOn status condition: %s", resource.ToJSON(gatewaycondition))
 
 		syncCtx.Recorder().Eventf("ManagedClusterAddOnStatusUpdated", "Updated status conditions:  %#v",
 			updatedStatus.Conditions)
 	}
 
+	routeAgentGVR := schema.GroupVersionResource{
+		Group:    "submariner.io",
+		Version:  "v1",
+		Resource: "routeagents",
+	}
+
+	routeAgentList, err := c.spokeDynamicClient.Resource(routeAgentGVR).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	var allUnhealthyMessages []string
+
+	for _, item := range routeAgentList.Items {
+		routeAgent := &submarinermv1.RouteAgent{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.UnstructuredContent(), &routeAgent); err != nil {
+			return err
+		}
+
+		routeAgentCondition := c.checkRouteAgentConditions(routeAgent)
+		if routeAgentCondition.Status == metav1.ConditionTrue && routeAgentCondition.Reason == connectionsDegraded {
+			allUnhealthyMessages = append(allUnhealthyMessages, routeAgentCondition.Message)
+		}
+	}
+
+	finalCondition := &metav1.Condition{
+		Type:   routeAgentConnectionDegraded,
+		Status: metav1.ConditionFalse,
+		Reason: "AllConnectionsHealthy",
+	}
+
+	if len(allUnhealthyMessages) > 0 {
+		finalCondition.Status = metav1.ConditionTrue
+		finalCondition.Reason = connectionsDegraded
+		finalCondition.Message = strings.Join(allUnhealthyMessages, "\n")
+	} else {
+		finalCondition.Message = "All RouteAgent connections to remote endpoints are established and healthy."
+	}
+
+	updatedStatus, updated, err = addon.UpdateStatus(ctx, c.addOnClient, c.clusterName, addon.UpdateConditionFn(finalCondition))
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		c.logger.Infof("Updated final RouteAgent ManagedClusterAddOn status condition: %s", resource.ToJSON(finalCondition))
+		syncCtx.Recorder().Eventf("ManagedClusterAddOnStatusUpdated", "Updated RouteAgent condition: %#v",
+			updatedStatus.Conditions)
+	}
```

Comment:
We can combine the two conditions in one `UpdateStatus` call:
```suggestion
	updatedStatus, updated, err = addon.UpdateStatus(ctx, c.addOnClient, c.clusterName, addon.UpdateConditionFn(gatewaycondition),
	addon.UpdateConditionFn(routeAgentCondition))
	if err != nil {
		return err
	}

	if updated {
		c.logger.Infof("Updated submariner ManagedClusterAddOn status condition: %s", resource.ToJSON(updatedStatus.Conditions))
		syncCtx.Recorder().Eventf("ManagedClusterAddOnStatusUpdated", "Updated status condition: %#v",
			updatedStatus.Conditions)
	}
```

Reply:
Done.